sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeFileSystemClient.cs

```diff
@@ -1581,5 +1581,272 @@ private void SetNameFieldsIfNull()
+                    conditions.ToBlobRequestConditions(),
+                    cancellationToken)
+                    .ConfigureAwait(false);
+
+                return Response.FromValue(
+                    response.Value.ToFileSystemAccessPolicy(),
+                    response.GetRawResponse());
+            }
+            catch (Exception ex)
+            {
+                scope.Failed(ex);
+                throw;
+            }
+            finally
+            {
+                scope.Dispose();
+            }
+        }
+        #endregion GetAccessPolicy
+
+        #region SetAccessPolicy
+        /// <summary>
+        /// The <see cref="SetAccessPolicy"/> operation sets the
+        /// permissions for the specified file system. The permissions indicate
+        /// whether file system data may be accessed publicly.
+        ///
+        /// For more information, see <see href=" https://docs.microsoft.com/rest/api/storageservices/set-container-acl" />.
+        /// </summary>
+        /// <param name="accessType">
+        /// Optionally specifies whether data in the file system may be accessed
+        /// publicly and the level of access. <see cref="Models.PublicAccessType.FileSystem"/>
+        /// specifies full public read access for file system and path data.
+        /// Clients can enumerate paths within the file system via anonymous
+        /// request, but cannot enumerate file systems within the storage
+        /// account.  <see cref="Models.PublicAccessType.Path"/> specifies public
+        /// read access for paths.  Path data within this file system can be
+        /// read via anonymous request, but file system data is not available.
+        /// Clients cannot enumerate paths within the file system via anonymous
+        /// request.  <see cref="Models.PublicAccessType.None"/> specifies that the
+        /// file system data is private to the account owner.
+        /// </param>
+        /// <param name="permissions">
+        /// Stored access policies that you can use to provide fine grained
+        /// control over container permissions.
+        /// </param>
+        /// <param name="conditions">
+        /// Optional <see cref="DataLakeRequestConditions"/> to add
+        /// conditions on setting this file systems's access policy.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Response{FileSystemInfo}"/> describing the
+        /// updated container.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="RequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        public virtual Response<FileSystemInfo> SetAccessPolicy(
```

Comment:
Can the body of this method be factored out into SetAccessPolicyInternal, similar to what BlobContainerClient has?
Also applies for GetAccessPolicy.

Reply:
For DataLake APIs that call into Blob, the Async API calls the public async blob API, and the sync API called the public sync API.

I'd prefer to only call public Blob APIs in DataLake.  Having internal APIs won't save us very much code, and would be an extensive refactor to apply this everywhere.  I think there are higher priority things we could look at.

Reply:
The main thing that it would help with is avoiding the duplicate exception handling and scope stuff, but I would agree it is not high priority.

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/8843.  If we every have a slow day, I'll loop back around to this one.

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/8843